### ISSUE TITLE
Investigate ComplementAlphabet replacement

### ISSUE THREAD

**Issue by rhagenson** [pony-bio/pony-bio#8]
I do not currently like (but see no immediate solution to) the current use of `ComplementAlphabet` for DNA and RNA. The `complement(...)` method must exist on the implementer of the alphabet itself so we cannot push the concern down to the letters themselves (problem: in DNA A->T, in RNA A->U). 

I would like to see: 1) unification toward the approach used by `Gapped` where stating the alphabet is complement-able is done via a type alias, or 2) determine a manner to have default behavior so a simple `is Complementable` approach can be used on definition.

**Comment by adri326** [comment]
The approach introduced with the first implementation is the closest alternative to rust's "conditional inheritance" paradigm:

```rust
impl Complementable for Gapped[T]
where T: Complementable & Alphabet {/* ... */ }
```

The above code would expose `Complementable` on Gapped iff T is Complementable and is an Alphabet.

This feature does not exist in Pony, and I naïvely faked it by duplicating code. Such duplication will be amplified down the line and make the code unmaintainable, thus the need to redo this implementation.

I am more in favor of option 2, which would allow for easier integration of `Gapped` with custom alphabets. I envision it like this:
- rename the `ComplementAlphabet` `Complement` and disconnect it from `Alphabet`
- give to the `complement` method a default identity implementation
- change `Gapped`'s generics (potentially addressing #11 at the same time)
- move the `complement` method from `ComplementGapped` to `Gapped` and tweak it accordingly
- remove `ComplementGapped`
- flag `Gapped` as `Complement[(T | GapType)]`

**Comment by rhagenson** [comment]
Fixed in #16
